lib/signs_ui_web/controllers/messages_controller.ex

```diff
@@ -87,4 +87,97 @@ defmodule SignsUiWeb.MessagesController do
   def create(conn, _params) do
     send_resp(conn, 201, "Ignoring unknown message.")
   end
+
+  def background(conn, _params) do
+    with {:ok, visual_zones} <- parse_zones(conn, "visual_zones"),
+         {:ok, visual_data} <- parse_visual_data(conn),
+         {:ok, expiration} <- parse_expiration(conn) do
+      scu_id =
+        Enum.find_value(conn.req_headers, fn {key, value} ->
+          if(key == "x-scu-id", do: value)
+        end)
```

Comment:
**suggestion:** i think [`get_req_header/2`](https://hexdocs.pm/plug/1.16.1/Plug.Conn.html#get_req_header/2) will do this for ya

```suggestion
      [scu_id | _] = get_req_header(conn, "x-scu-id")
```

Reply:
Ah, thanks, I didn't know about that.